rotor/storage.nelua

```diff
@@ -35,9 +35,13 @@ require 'vector'
    end
 
    function StorageT:new_entry(entry_content: T): (GenerationalIndex, T*)
+      if #self.next_free_ids == 0 then
+         self.next_free_ids:push(GenerationalIndex.new(0, 0))
+      end
       local last_free_id = self.next_free_ids[#self.next_free_ids - 1]
 
       self.entries:insert(last_free_id.index, entry_content)
+      local entry: T* = &self.entries[last_free_id.index]
```

Comment:
This gets the correct reference for the entry, before the fix a reference to a temporary value was being returned and would cause crash. In the future this kind of bug may be triggered by the compiler, but right now the user should take care when using references to temporaries.

This fix is not fully safe, the returned reference is valid until other `self.entries:insert` is made, this means that if the returned reference is stored elsewhere in the code it will become invalid once a new entry is added to the entries vector.